extensions/prototype-content-with-issues-table/app.R

```diff
@@ -119,7 +90,34 @@ server <- function(input, output, session) {
   
   # output the datatable of failed jobs
   output$jobs <- renderDT({
-    datatable(bad_content_df(), 
+    datatable(bad_content_df() |>
+                # map job type to something more readable
+                mutate(failed_job_type = case_when(
+                  failed_job_type %in% c("build_report", "build_site", "build_jupyter") ~ "Building",
+                  failed_job_type %in% c("packrat_restore", "python_restore") ~ "Restoring environment",
+                  failed_job_type == "configure_report" ~ "Configuring report",
+                  failed_job_type %in% c("run_app", 
+                             "run_api", 
+                             "run_tensorflow", 
+                             "run_python_api",
+                             "run_dash_app",
+                             "run_gradio_app",
+                             "run_streamlit",
+                             "run_bokeh_app",
+                             "run_fastapi_app",
+                             "run_voila_app",
+                             "run_pyshiny_app") ~ "Running",
+                  failed_job_type == "render_shiny" ~ "Rendering",
+                  failed_job_type == "ctrl_extraction" ~ "Extracting parameters",
+                  TRUE ~ failed_job_type)) |>
+                # map exit codes to something more readable 
+                mutate(failure_reason = case_when(
+                  failure_reason %in% c(1, 2, 134) ~ "failed to run / error during running",
+                  failure_reason == 137 ~ "out of memory",
+                  failure_reason %in% c(255, 15, 130) ~ "process terminated by server",
+                  failure_reason %in% c(13, 127) ~ "configuration / permissions error",
+                  TRUE ~ as.character(failure_reason))) |>
```

Comment:
In response to:

> [Do we need to turn this into characters at all?](https://github.com/posit-dev/connect-extensions/pull/33#discussion_r1989912855)

We did not need to convert prior to `mutate()` but still need to treat all unmapped `exit_code` values as characters, or there will be a type mismatch.

Reply:
_nods_ great, this might be worth noting "treat all unmapped exit_code values as characters for type consistency" in a comment above line 119 here